mbed/mbed.py

```diff
@@ -2977,7 +2982,7 @@ def test_(toolchain=None, target=None, compile_list=False, run_list=False,
 def dev_mgmt(toolchain=None, target=None, source=False, profile=False, build=False):
     orig_path = getcwd()
     program = Program(getcwd(), True)
-    program.check_requirements(True)
+    program.check_requirements()
```

Comment:
This is a behavior change, this will now cause the CLI to halt and exit with a non-zero return code if the requirements aren't present before running the `mbed dm` command.

I'm not necessarily against this, it just wasn't documented in the PR description or the commit message so I missed this at first.

Reply:
To clarify the behavior:
* Mbed CLI should attempt to auto-install requirements when:
   * Codebased is being imported (`mbed import`, `mbed add`, `mbed deploy`, `mbed new`)
   * Codebased is being updated/checked out (`mbed update`)
   * Certificates are being installed or changed (`mbed dm`)
* Mbed CLI should only check and NOT try to auto-install requirements when:
   * Code is being compiled, tested or exported
   * Any Mbed CLI operation that doesn't imply network connectivity

As thoroughly described here https://github.com/ARMmbed/mbed-cli/issues/756. Also as mentioned in my first comment to this PR.

> Related to #795, #782 and #756